### DISCUSSION TITLE
Offer a coalescing operator

### DISCUSSION THREAD

**Discussion by PeachOS** [nlohmann/json#2715] [Ideas]
Sometimes it would be nice to have a default value operator in case a field is null. 
In case a field is missing, I can use `obj.value("field", 0.0)`. There is no pendant for null values though and I would have to write something like `auto val = obj["field"].is_null() ? 0.0 : obj["field"].get<double>()` which gets ugly when used for several fields.

I could imagine that an operator for such situation could be very handy. I would imagine something like the following:
```cpp
// inspired from SQL
double value = obj.ifnull("field", 0.0);  // if obj["field"] is null, return 0.0

// probably not much of a use case but why not mention the more generic SQL inspired operator
double value = obj.coalesce("field1", "field2", 0.0);  // takes the first non-null value among obj["field1"], obj["field2"], and 0.0
```

What do you think?

**Comment by nlohmann** [comment]
There's no `ifnull`/`coalesce`-style helper in the current API (a grep for 'coalesce'/'ifnull' in json.hpp finds nothing), so today this genuinely requires the ternary-with-is_null() pattern described. This overlaps heavily with the closely related #4245 (contains_not_null) and the general null-safe-access theme in #5129 — a single well-designed 'default if null or missing' accessor would likely satisfy all three at once, so it's worth consolidating rather than adding several overlapping convenience methods.

---
*This reply was drafted by [Claude Code](https://claude.com/claude-code) on behalf of @nlohmann, as part of a review of open discussions.*
